File: add_filters.js
const fs = require('fs');
const path = require('path');

const entityDir = 'd:\\EduConnect\\src\\main\\java\\com\\educonnect\\model';
const entities = ["User.java", "Student.java", "Teacher.java", "Parent.java", "Classroom.java", "Announcement.java", "Message.java"];

const importStr = "import org.hibernate.annotations.Filter;\nimport org.hibernate.annotations.FilterDef;\nimport org.hibernate.annotations.ParamDef;\n";
const annotationStr = `@FilterDef(name = "tenantFilter", parameters = @ParamDef(name = "schoolId", type = Long.class))\n@Filter(name = "tenantFilter", condition = "school_id = :schoolId")\n`;

for (const entity of entities) {
    const filePath = path.join(entityDir, entity);
    let content = fs.readFileSync(filePath, 'utf-8');

    if (content.includes('@FilterDef')) continue;

    content = content.replace(/(import jakarta\.persistence\.\*;)/, `$1\n${importStr}`);
    content = content.replace('@Entity', `${annotationStr}@Entity`);

    fs.writeFileSync(filePath, content, 'utf-8');
}
console.log('Entities updated.');

File: add_filters.py
import os
import re

entity_dir = r"d:\EduConnect\src\main\java\com\educonnect\model"
entities = ["User.java", "Student.java", "Teacher.java", "Parent.java", "Classroom.java", "Announcement.java", "Message.java"]

import_str = "import org.hibernate.annotations.Filter;\nimport org.hibernate.annotations.FilterDef;\nimport org.hibernate.annotations.ParamDef;\n"

annotation_str = """@FilterDef(name = "tenantFilter", parameters = @ParamDef(name = "schoolId", type = Long.class))
@Filter(name = "tenantFilter", condition = "school_id = :schoolId")
"""

for entity in entities:
    path = os.path.join(entity_dir, entity)
    with open(path, "r", encoding="utf-8") as f:
        content = f.read()
    
    if "@FilterDef" in content:
        continue

    # Add imports
    content = re.sub(r'(import jakarta\.persistence\.\*;)', r'\1\n' + import_str, content)
    
    # Add annotations before @Entity
    content = content.replace("@Entity", annotation_str + "@Entity")
    
    with open(path, "w", encoding="utf-8") as f:
        f.write(content)

print("Entities updated.")
